code/build.R

```diff
@@ -63,69 +62,69 @@ if (length(failed_regions) > 0) {
   message("All regions processed successfully on the first attempt.")
 }
 
-# Zip the contents of the outputdata folder
-output_folder = file.path("outputdata", parameters$date_routing)
-list.files(output_folder)
-setwd("outputdata")
-zip_file = paste0(parameters$date_routing, ".zip")
-zip(zipfile = zip_file, parameters$date_routing, extras = "-x *.Rds")
 
+# Zip and upload outputs ----
+setwd("outputdata") # Temporarily change working directory
 date_folder = parameters$date_routing
-
-dir.create(paste0(date_folder, "-geojson"))
-
-for(i in list.dirs(date_folder)) {
-  r = gsub(pattern = paste0(date_folder, "/"), replacement = "", x = i)
-  f = list.files(i, pattern = "geojson", full.names = TRUE)
-  f_new = file.path(paste0(date_folder, "-geojson"), paste0(r, "_", basename(f)))
-  message(paste(f, collapse = "\n"))
-  message(paste(f_new, collapse = "\n"))
-  file.copy(f, f_new)
-}
-
+list.files(date_folder)
+zip_file = paste0(date_folder, ".zip")
+zip(zipfile = zip_file, date_folder, extras = "-x *.Rds")
 zip(paste0(date_folder, "-geojson.zip"), files = paste0(date_folder, "-geojson"))
 fs::file_size(paste0(date_folder, "-geojson.zip"))
 system("gh release list")
 system(paste0("gh release create ", date_folder, "-geojson"))
 system(paste0("gh release upload ", date_folder, "-geojson ", date_folder, "-geojson.zip"))
+setwd("..")
+
+region = region_names[1] # for testing
+
+# Generate coherent network ---------------------------------------------------
+remotes::install_github("nptscot/corenet")
 
-# Generate coherent network
 for (region in region_names[1:6]) {
     message("Processing coherent network for region: ", region)
-    parameters$region = region
-    parameters$coherent_area = cities_region_names[[region]]
+    region_snake = snakecase::to_snake_case(region)
+    coherent_area = cities_region_names[[region]]
 
-    combined_network_tile_path = paste0("outputdata/", parameters$date_routing,"/",  snakecase::to_snake_case(parameters$region), "/", "combined_network_tile.geojson")
-    NPT = sf::read_sf(combined_network_tile_path) |>
+    cnet_path = file.path(output_folder, region_snake, "combined_network_tile.geojson")
+    combined_net = sf::read_sf(cnet_path) |>
       sf::st_transform(crs = "EPSG:27700")
 
-    folder_path = paste0("outputdata/", parameters$date_routing,"/",  snakecase::to_snake_case(parameters$region), "/", "coherent_networks/")
+    folder_path = file.path(output_folder, region_snake, "coherent_networks")
 
     if(!dir.exists(folder_path)) {
       dir.create(folder_path, recursive = TRUE)
     }
-
-    for (city in parameters$coherent_area) { 
+    
+    open_roads_path = paste0("inputdata/OS_Scotland_Network_", region_snake, ".geojson")
```

Comment:
Heads-up @wangzhao0217 this is the cause. Where is the input data?

It only has 2 columns:

```
>     open_roads_path = paste0("inputdata/OS_Scotland_Network_", region_snake, ".geojson")
>     open_roads_unprojected = sf::read_sf(open_roads_path)
>     open_roads = sf::st_transform(open_roads_unprojected, "EPSG:27700")
>     names(open_roads)
[1] "identifier" "geometry"